src/rpc/types.ts

```diff
@@ -84,7 +85,7 @@ type JsonRpcResponseError = {
 //  -32603	Internal error	Internal JSON-RPC error.
 //  -32000 to -32099
```

Comment:
I don't think we should be using these codes.

Instead you should just have unique exception classes for each possible error.

The reason is because in an OOP language like ourselves, the class name itself is sufficient identification, code is unnecessary.

We do use `code` for sysexit codes, but that's only because a number is necessary when exiting a program.

When sending the error over the RPC, we don't need it, we only need the serialised version of the exception object which is then later deserialised on the remote side.

Reply:
So you should just literally have:

```
class ErrorRPCMethodNotFound { }
class ErrorRPCInvalidParams { }
```

Reply:
If the JSON RPC spec requires a `code: number`, we can fill it with our sysexit code. But the really we should be replacing that with our "serialised" error structure. The one we use with `fromJSON` and `toJSON`.

Reply:
We're not using these codes anywhere. They were for reference. They can be removed.